[MRG] Use return_X_y=True with load_boston where appropriate

allenakinkunle opened the pull request:
#### Reference Issues/PRs
Partially addresses #14347 (Use return_X_y=True when applicable in examples)

#### What does this implement/fix? Explain your changes.
Changes how `load_boston` is used in examples and tests where applicable.

#### Any other comments?
*

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

allenakinkunle committed:
Use return_X_y=True with load_boston where appropriate

qinhanmin2014 commented on examples/applications/plot_model_complexity_influence.py:
```diff
@@ -114,14 +114,15 @@ def plot_influence(conf, mse_values, prediction_times, complexities):
     host.axis["left"].label.set_color(p1.get_color())
     par1.axis["right"].label.set_color(p2.get_color())
-    plt.title('Influence of Model Complexity - %s' % conf['estimator'].__name__)
+    plt.title('Influence of Model Complexity: %s' % conf['estimator'].__name__)
```
please avoid irrelevant changes.

allenakinkunle replied:
Duly noted.

qinhanmin2014 commented on examples/applications/plot_model_complexity_influence.py:
```diff
@@ -114,14 +114,15 @@ def plot_influence(conf, mse_values, prediction_times, complexities):
     a = estimator.coef_.toarray()
     return np.count_nonzero(a)
 
+
```
please avoid irrelevant changes.

adrinjalali commented:
@allenakinkunle once you address @qinhanmin2014 's comments, this is good to be merged :)

allenakinkunle commented:
> @allenakinkunle once you address @qinhanmin2014 's comments, this is good to be merged :)

@adrinjalali I have addressed the comments. Thank you for reviewing.
